client/app/bundles/course/assessment/submission/components/ReadOnlyEditor/NarrowEditor.jsx

```diff
@@ -171,8 +173,14 @@ export default function NarrowEditor(props) {
     props.toggleLine(lineNumber);
   };
 
-  const collapseComment = (lineNumber) => {
-    props.collapseLine(lineNumber);
+  const collapseComment = (lineNumber, event) => {
+    // CKEditor's Link popup dialog is rendered separately in a separate wrapper (ck-body-wrapper)
+    // and not rendered as a child of the main CKEditor's toolbar.
+    // As a result, the clickawaylistener would be triggered when the Link popup dialog is clicked.
+    // Here, we check the class' of the clicked element and if contains "ck", the comment is not collapsed.
+    // There is a downside to this that lets say if another ckeditor toolbar is clicked, the comment is also
+    // not collapsed, however, this is not a big issue as the former issue would be more disruptive for users.
+    if (!event.target.classList.contains('ck')) props.collapseLine(lineNumber);
```

Comment:
https://stackoverflow.com/a/56807169

> You can use the closest() method of Element that traverses parents (heading toward the document root) of the Element until it finds a node that matches the provided selectorString. Will return itself or the matching ancestor. If no such element exists, it returns null.
> 
> You can convert the returned value into boolean

_However, it has no support for an ancient browser called Internet Explorer, and would require a polyfill._

Simply checking for the `.ck` class feels like it might cause unexpected issues if we are not careful.
Can we target the separate wrapper `ck-body-wrapper`?
`if (Boolean(event.target.closest(".ck-body-wrapper")))` and prevent default/early return instead of this.

Reply:
Somehow, it doesnt work for every button inside the link dialog. So we will stick to the current one now.